vips/foreign.c

```diff
@@ -1,29 +1,92 @@
 #include "lang.h"
 #include "foreign.h"
+#include "govips.h"
 
-int load_image_buffer(void *buf, size_t len, int imageType, VipsImage **out) {
+int load_image_buffer_operation(VipsOperation *operation, VipsBlob *blob, VipsImage **out, ImportParams *params) {
+    int result;
+
+    // Set operation parameters
+    result = vips_object_set(VIPS_OBJECT(operation), "buffer", blob, "out", out, NULL);
+    if (result) {
+        return result;
+    }
+
+    if (params->fail_set) {
```

Comment:
This repeating pattern can be DRYed up in a helper function to handle logging if it fails.

But when does this sort of things fail?

Reply:
I can use `#define` for this block if you want to.

Reply:
Yeah, that may be better.